presto-product-tests/src/main/resources/sql-tests/testcases/window_functions/windowSpecificationBcpAA.sql

```diff
@@ -0,0 +1,8 @@
+-- database: presto; groups: window;
+select
+suppkey, orderkey, partkey,
+nth_value(quantity, 4) over (partition by partkey order by orderkey rows between UNBOUNDED preceding and CURRENT ROW) nth_value_quantity_B,
+sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A,
+sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A
```

Comment:
ORDER BY looks good - it produces the same basic unoptimized plan as when you don't include ORDER BY

GROUP BY produces the error ``` GROUP BY clause cannot contain aggregations or window functions: [...```

HAVING generates an error: ```sum(double):double is not a scalar function```, which seems like it might be a cryptic error meaning the same thing?

The postgresql [docs](https://www.postgresql.org/docs/current/static/sql-expressions.html#SYNTAX-WINDOW-FUNCTIONS) indicate that window functions can only be used in SELECT and ORDER BY.

The 2011 spec says "The &lt;search condition> shall not contain a &lt;window function> without an intervening &lt;query expression>." I'll try to come up with a conforming query and see what happens :-)